sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DoFnReflectorTest.java

```diff
@@ -534,102 +559,148 @@ public void testBadReturnType() throws Exception {
-      WindowingInternals<Integer, String> i1) {}
+  private void goodGenerics(
+      DoFn<Integer, String>.ProcessContext c,
+      DoFn.InputProvider<Integer> input,
+      DoFn.OutputReceiver<String> output) {}
 
   @Test
   public void testValidGenerics() throws Exception {
-    Method method = getClass().getDeclaredMethod("goodGenerics",
-        DoFn.ProcessContext.class, WindowingInternals.class);
+    Method method =
+        getClass()
+            .getDeclaredMethod(
+                "goodGenerics",
+                DoFn.ProcessContext.class,
+                DoFn.InputProvider.class,
+                DoFn.OutputReceiver.class);
     DoFnReflector.verifyProcessMethodArguments(method);
   }
 
   @SuppressWarnings("unused")
-  private void goodWildcards(DoFn<Integer, String>.ProcessContext c,
-      WindowingInternals<?, ?> i1) {}
+  private void goodWildcards(
+      DoFn<Integer, String>.ProcessContext c,
+      DoFn.InputProvider<?> input,
+      DoFn.OutputReceiver<?> output) {}
 
   @Test
   public void testGoodWildcards() throws Exception {
-    Method method = getClass().getDeclaredMethod("goodWildcards",
-        DoFn.ProcessContext.class, WindowingInternals.class);
+    Method method =
+        getClass()
+            .getDeclaredMethod(
+                "goodWildcards",
+                DoFn.ProcessContext.class,
+                DoFn.InputProvider.class,
+                DoFn.OutputReceiver.class);
     DoFnReflector.verifyProcessMethodArguments(method);
   }
 
   @SuppressWarnings("unused")
-  private void goodBoundedWildcards(DoFn<Integer, String>.ProcessContext c,
-      WindowingInternals<? super Integer, ? super String> i1) {}
+  private void goodBoundedWildcards(
+      DoFn<Integer, String>.ProcessContext c,
+      DoFn.InputProvider<? super Integer> input,
+      DoFn.OutputReceiver<? super String> output) {}
 
   @Test
   public void testGoodBoundedWildcards() throws Exception {
-    Method method = getClass().getDeclaredMethod("goodBoundedWildcards",
-        DoFn.ProcessContext.class, WindowingInternals.class);
+    Method method =
+        getClass()
+            .getDeclaredMethod(
+                "goodBoundedWildcards",
+                DoFn.ProcessContext.class,
+                DoFn.InputProvider.class,
+                DoFn.OutputReceiver.class);
     DoFnReflector.verifyProcessMethodArguments(method);
   }
 
   @SuppressWarnings("unused")
   private <InputT, OutputT> void goodTypeVariables(
       DoFn<InputT, OutputT>.ProcessContext c,
-      WindowingInternals<InputT, OutputT> i1) {}
+      DoFn.InputProvider<InputT> input,
+      DoFn.OutputReceiver<OutputT> output) {}
 
   @Test
   public void testGoodTypeVariables() throws Exception {
-    Method method = getClass().getDeclaredMethod("goodTypeVariables",
-        DoFn.ProcessContext.class, WindowingInternals.class);
+    Method method =
+        getClass()
+            .getDeclaredMethod(
```

Comment:
Used [google-java-format](https://github.com/Google/google-java-format). It has a functional IntelliJ plugin and formats most things quite nicely. Easier to go autoformat than to discuss it :-)